src/modules/paste.ts

```diff
@@ -25,24 +25,25 @@ export class PasteEvent extends Event {
   }
 }
 
+export interface PasteOptions {
+  text?: string;
+  html?: string;
+  selection?: EditorRange | null;
+}
+
 export interface PasteOptions {
   htmlParser?: (editor: Editor, html: string) => Delta;
 }
 
 export function paste(editor: Editor, options?: PasteOptions) {
 
-  function onPaste(event: ClipboardEvent) {
-    if (!editor.enabled || !editor.doc.selection) return;
-    event.preventDefault();
-    const dataTransfer = event.clipboardData;
+  function paste({ selection, text, html }: PasteOptions) {
     const { doc } = editor;
-    const selection = doc.selection && normalizeRange(doc.selection);
-    if (!dataTransfer || !selection) return;
+    selection = selection || doc.selection;
+    selection = selection && doc.selection && normalizeRange(doc.selection);
```

Comment:
This line should be
```ts
selection = selection && normalizeRange(selection);
```

otherwise the `options.selection` parameter is ignored and only `doc.selection` is used.

Reply:
fixed.